<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mAMainWindow</class>
 <widget class="QMainWindow" name="mAMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>620</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>300</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>miniAudicle</string>
  </property>
  <property name="windowIcon">
   <iconset resource="miniAudicle.qrc">
    <normaloff>:/icon/miniAudicle.png</normaloff>:/icon/miniAudicle.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>581</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuRecent_Files">
     <property name="title">
      <string>Open Recent</string>
     </property>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionOpen_Example"/>
    <addaction name="menuRecent_Files"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionExport_as_WAV"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuChucK">
    <property name="title">
     <string>ChucK</string>
    </property>
    <widget class="QMenu" name="menuLog_Level">
     <property name="title">
      <string>Log Level</string>
     </property>
     <addaction name="actionLogNone"/>
     <addaction name="actionLogCore"/>
     <addaction name="actionLogSystem"/>
     <addaction name="actionLogSevere"/>
     <addaction name="actionLogWarning"/>
     <addaction name="actionLogInfo"/>
     <addaction name="actionLogConfig"/>
     <addaction name="actionLogFine"/>
     <addaction name="actionLogFiner"/>
     <addaction name="actionLogFinest"/>
     <addaction name="actionLogCrazy"/>
    </widget>
    <addaction name="actionAdd_Shred"/>
    <addaction name="actionReplace_Shred"/>
    <addaction name="actionRemove_Shred"/>
    <addaction name="separator"/>
    <addaction name="actionAdd_All_Open_Documents"/>
    <addaction name="actionReplace_All_Open_Documents"/>
    <addaction name="actionRemove_All_Open_Documents"/>
    <addaction name="separator"/>
    <addaction name="actionRemove_Last_Shred"/>
    <addaction name="actionRemove_All_Shreds"/>
    <addaction name="actionClear_Virtual_Machine"/>
    <addaction name="separator"/>
    <addaction name="actionAbort_Currently_Running_Shred"/>
    <addaction name="separator"/>
    <addaction name="actionStart_Virtual_Machine"/>
    <addaction name="menuLog_Level"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionConsole_Monitor"/>
    <addaction name="actionVirtual_Machine_Monitor"/>
    <addaction name="actionDevice_Browser"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionMiniAudicle_website"/>
    <addaction name="actionChucK_website"/>
    <addaction name="separator"/>
    <addaction name="actionAbout_miniAudicle"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuChucK"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionAdd_Shred"/>
   <addaction name="actionReplace_Shred"/>
   <addaction name="actionRemove_Shred"/>
   <addaction name="actionRemove_Last_Shred"/>
   <addaction name="actionClear_Virtual_Machine"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="toolTip">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="toolTip">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences...</string>
   </property>
   <property name="toolTip">
    <string>Preferences...</string>
   </property>
   <property name="shortcut">
    <string>Alt+,</string>
   </property>
  </action>
  <action name="actionAdd_Shred">
   <property name="icon">
    <iconset>
     <normalon>:/icon/add.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Add Shred</string>
   </property>
   <property name="shortcut">
    <string>Alt++</string>
   </property>
  </action>
  <action name="actionReplace_Shred">
   <property name="icon">
    <iconset resource="miniAudicle.qrc">
     <normaloff>:/icon/replace.png</normaloff>:/icon/replace.png</iconset>
   </property>
   <property name="text">
    <string>Replace Shred</string>
   </property>
   <property name="shortcut">
    <string>Alt+=</string>
   </property>
  </action>
  <action name="actionRemove_Shred">
   <property name="icon">
    <iconset resource="miniAudicle.qrc">
     <normaloff>:/icon/remove.png</normaloff>:/icon/remove.png</iconset>
   </property>
   <property name="text">
    <string>Remove Shred</string>
   </property>
   <property name="shortcut">
    <string>Alt+-</string>
   </property>
  </action>
  <action name="actionAdd_All_Open_Documents">
   <property name="text">
    <string>Add All Open Documents</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+Shift+=</string>
   </property>
  </action>
  <action name="actionReplace_All_Open_Documents">
   <property name="text">
    <string>Replace All Open Documents</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+=</string>
   </property>
  </action>
  <action name="actionRemove_All_Open_Documents">
   <property name="text">
    <string>Remove All Open Documents</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+-</string>
   </property>
  </action>
  <action name="actionRemove_All_Shreds">
   <property name="text">
    <string>Remove All Shreds</string>
   </property>
  </action>
  <action name="actionRemove_Last_Shred">
   <property name="icon">
    <iconset resource="miniAudicle.qrc">
     <normaloff>:/icon/removelast.png</normaloff>:/icon/removelast.png</iconset>
   </property>
   <property name="text">
    <string>Remove Last Shred</string>
   </property>
  </action>
  <action name="actionStart_Virtual_Machine">
   <property name="text">
    <string>Start Virtual Machine</string>
   </property>
   <property name="shortcut">
    <string>Alt+.</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionAbort_Currently_Running_Shred">
   <property name="text">
    <string>Abort Currently Running Shred</string>
   </property>
   <property name="shortcut">
    <string>Alt+K</string>
   </property>
  </action>
  <action name="actionVirtual_Machine_Monitor">
   <property name="text">
    <string>Virtual Machine Monitor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="actionConsole_Monitor">
   <property name="text">
    <string>Console Monitor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="actionMiniAudicle_website">
   <property name="text">
    <string>miniAudicle website...</string>
   </property>
   <property name="toolTip">
    <string>miniAudicle website...</string>
   </property>
  </action>
  <action name="actionChucK_website">
   <property name="text">
    <string>ChucK website...</string>
   </property>
   <property name="toolTip">
    <string>ChucK website...</string>
   </property>
  </action>
  <action name="actionAbout_miniAudicle">
   <property name="text">
    <string>About miniAudicle</string>
   </property>
  </action>
  <action name="actionLogNone">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>None</string>
   </property>
  </action>
  <action name="actionLogCore">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Core</string>
   </property>
  </action>
  <action name="actionLogSystem">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>System</string>
   </property>
  </action>
  <action name="actionLogSevere">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Severe</string>
   </property>
  </action>
  <action name="actionLogWarning">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Warning</string>
   </property>
  </action>
  <action name="actionLogInfo">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Info</string>
   </property>
  </action>
  <action name="actionLogConfig">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Config</string>
   </property>
  </action>
  <action name="actionLogFine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Fine</string>
   </property>
  </action>
  <action name="actionLogFiner">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Finer</string>
   </property>
  </action>
  <action name="actionLogFinest">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Finest</string>
   </property>
  </action>
  <action name="actionLogCrazy">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Crazy</string>
   </property>
  </action>
  <action name="actionRecent">
   <property name="text">
    <string>Recent</string>
   </property>
  </action>
  <action name="actionOpen_Example">
   <property name="text">
    <string>Open Example...</string>
   </property>
  </action>
  <action name="actionExport_as_WAV">
   <property name="text">
    <string>Export...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+E</string>
   </property>
  </action>
  <action name="actionClear_Virtual_Machine">
   <property name="icon">
    <iconset resource="miniAudicle.qrc">
     <normaloff>:/icon/removeall.png</normaloff>:/icon/removeall.png</iconset>
   </property>
   <property name="text">
    <string>Clear Virtual Machine</string>
   </property>
   <property name="toolTip">
    <string>Clear Virtual Machine</string>
   </property>
   <property name="shortcut">
    <string>Alt+Del, Ctrl+S</string>
   </property>
  </action>
  <action name="actionDevice_Browser">
   <property name="text">
    <string>Device Browser</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+2</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="miniAudicle.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>exit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>newFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>closeFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>mAMainWindow</receiver>
   <slot>closeFile(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>saveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStart_Virtual_Machine</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>toggleVM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_Shred</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>addCurrentDocument()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemove_Shred</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>removeCurrentDocument()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReplace_Shred</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>replaceCurrentDocument()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemove_Last_Shred</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>removeLastShred()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemove_All_Shreds</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>removeAllShreds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout_miniAudicle</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConsole_Monitor</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>showConsoleMonitor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVirtual_Machine_Monitor</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>showVirtualMachineMonitor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogConfig</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogCore</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogCrazy</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogFine</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogFiner</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogFinest</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogInfo</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogNone</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogSevere</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogSystem</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLogWarning</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>setLogLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>showPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_Example</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>openExample()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>currentChanged(int)</signal>
   <receiver>mAMainWindow</receiver>
   <slot>tabSelected(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbort_Currently_Running_Shred</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>abortCurrentShred()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear_Virtual_Machine</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>clearVM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDevice_Browser</sender>
   <signal>triggered()</signal>
   <receiver>mAMainWindow</receiver>
   <slot>showDeviceBrowser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>exit()</slot>
  <slot>openFile()</slot>
  <slot>newFile()</slot>
  <slot>closeFile()</slot>
  <slot>closeFile(int)</slot>
  <slot>saveFile()</slot>
  <slot>toggleVM()</slot>
  <slot>addCurrentDocument()</slot>
  <slot>replaceCurrentDocument()</slot>
  <slot>removeCurrentDocument()</slot>
  <slot>removeLastShred()</slot>
  <slot>removeAllShreds()</slot>
  <slot>about()</slot>
  <slot>showConsoleMonitor()</slot>
  <slot>showVirtualMachineMonitor()</slot>
  <slot>setLogLevel()</slot>
  <slot>showPreferences()</slot>
  <slot>openExample()</slot>
  <slot>saveAs()</slot>
  <slot>tabSelected(int)</slot>
  <slot>abortCurrentShred()</slot>
  <slot>clearVM()</slot>
  <slot>showDeviceBrowser()</slot>
 </slots>
</ui>
